daffodil-cli/bin.NOTICE

```diff
@@ -11,118 +11,136 @@ Based on source code originally developed by
 
 The following NOTICE information applies to binary components distributed with this project:
 
-Apache Commons IO (lib/commons-io.commons-io-<VERSION>.jar)
-  Apache Commons IO
-  Copyright 2002-2021 The Apache Software Foundation
+Jackson JSON processor (lib/com.fasterxml.jackson.core.jackson-core-<VERSION>.jar)
```

Comment:
The new ordering isn't random.  Rather, it reorders all the sub-NOTICEs by their jar name so all the sub-NOTICEs are in the same order as a listing of the jars in the lib directory.  This serves three purposes: 1) we can notice immediately when new jars haven't been added to the sub-NOTICEs yet; 2) we know unambiguously where to insert new sub-NOTICEs to keep them ordered by jar name; and 2) we can methodically go through an Emacs dired buffer listing the jars in the lib directory, look inside every jar, and update the sub-NOTICEs if we do miss some changes since the majority of jars already have NOTICE files packaged inside the jars.